codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -262,11 +281,11 @@ public static void addHttp2ToHttpHeaders(int streamId, Http2Headers sourceHeader
             throw streamError(streamId, PROTOCOL_ERROR, t, "HTTP/2 to HTTP/1.x headers conversion error");
         }
 
-        headers.remove(HttpHeaderNames.TRANSFER_ENCODING);
-        headers.remove(HttpHeaderNames.TRAILER);
-        if (!addToTrailer) {
-            headers.setInt(ExtensionHeaderNames.STREAM_ID.text(), streamId);
-            HttpUtil.setKeepAlive(destinationMessage, true);
+        outputHeaders.remove(HttpHeaderNames.TRANSFER_ENCODING);
+        outputHeaders.remove(HttpHeaderNames.TRAILER);
+        if (!isTrailer) {
+            outputHeaders.setInt(ExtensionHeaderNames.STREAM_ID.text(), streamId);
+            HttpUtil.setKeepAlive(outputHeaders, httpVersion, true);
```

Comment:
@normanmaurer - Can you elaborate? Here is the signature of the method being called:

```java
void io.netty.handler.codec.http.HttpUtil.setKeepAlive(HttpHeaders h, HttpVersion httpVersion, boolean keepAlive);
```

I don't see where the boxing occurs.